src/odbc_connection.cpp

```diff
@@ -2634,6 +2737,47 @@ bind_buffers
     // bind depending on the column
     switch(column->DataType) {
 
+      // LONG data types should be retrieved through SQLGetData and not
+      // SQLBindCol/SQLFetch, as the buffers for SQLBindCol could be absurd
+      // sizes for small amounts of data transferred. However, if the fetch
+      // size is greater than 1 and the user's driver doesn't support SQLGetData
+      // with block cursors, we need to bite the bullet and bind with
+      // SQLBindCol.
+      case SQL_WLONGVARCHAR:
+        column->bind_type = SQL_C_WCHAR;
+        column->is_long_data = true;
```

Comment:
I think you need to put this in an else block or something, since the SQLBindCol call below has an `if (!is_long_data)` check surrounding it.

Reply:
Good call. I think I can remove a check when fetching data then, since `is_long_data` can just be used to track if:
* Multiple rows were fetched and block cursor is supported, or
* One row was fetched, in which case SQLGetData should work fine

Reply:
Latest changes look good, but is `is_long_data` ever set to `false`?

Reply:
It looks like I just rely on the struct value being initialized to 0, but that isn't a safe assumption (though it has been working in my cases). Will initialize to false.